cadcdata/cadcdata/core.py

```diff
@@ -0,0 +1,418 @@
+#  License, or (at your option)         licence, soit (à votre gré)                                                                                        
+#  any later version.                   toute version ultérieure.                                                                                          
+#                                                                                                                                                          
+#  OpenCADC is distributed in the       OpenCADC est distribué                                                                                             
+#  hope that it will be useful,         dans l’espoir qu’il vous                                                                                           
+#  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE                                                                                       
+#  without even the implied             GARANTIE : sans même la garantie                                                                                   
+#  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ                                                                                   
+#  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF                                                                                      
+#  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence                                                                                  
+#  General Public License for           Générale Publique GNU Affero
+#  more details.                        pour plus de détails.
+#                                       
+#  You should have received             Vous devriez avoir reçu une
+#  a copy of the GNU Affero             copie de la Licence Générale
+#  General Public License along         Publique GNU Affero avec
+#  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+#  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+#                                       <http://www.gnu.org/licenses/>.
+#
+#  $Revision: 4 $
+#
+# ***********************************************************************
+#
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import argparse
+import imp
+import logging
+import os
+import os.path
+import sys
+from StringIO import StringIO
+from datetime import datetime
+
+from cadcutils import net, util, exceptions
+from caom2.obs_reader_writer import ObservationReader, ObservationWriter
+from caom2.version import version as caom2_version
+from six.moves.urllib.parse import urlparse
+import zlib
+
+# from . import version as caom2repo_version
+from cadcdata import version
+
+__all__ = ['CadcDataClient']
+
+# TODO replace with SERVICE_URI when server supports it
+SERVICE_URL = 'www.cadc-ccda.hia-iha.nrc-cnrc.gc.ca/'
```

Comment:
This is going to be replaced in this iteration by a more appropriate registry-like look up so I won't bother with it right now.